### PR TITLE
enable scala-native junit tests

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -121,6 +121,21 @@ def configureJUnit(crossProject: CrossProject) = {
     libraryDependencies +=
       "com.github.sbt" % "junit-interface" % "0.13.3" % "test"
   )
+  .nativeSettings(
+    libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion,
+    addCompilerPlugin("org.scala-native" % "junit-plugin" % nativeVersion cross CrossVersion.full),
+    pomPostProcess := { node =>
```
Comment (xuwei-k) [inline build.sbt:127]: - https://github.com/scala-native/scala-native/issues/2540

File: build.sbt [hunk 2]
```diff
@@ -121,6 +121,21 @@ def configureJUnit(crossProject: CrossProject) = {
     libraryDependencies +=
       "com.github.sbt" % "junit-interface" % "0.13.3" % "test"
   )
+  .nativeSettings(
+    libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion,
```
Comment (joroKr21) [inline build.sbt:125]: Why not `% Test`?

Comment (xuwei-k) [reply build.sbt:125]: scala-native compiler plugin( `"org.scala-native" % "junit-plugin"` ) throw error in compile time if set `% Test`.

```
[error] scala.reflect.internal.MissingRequirementError: object org.junit.Test in compiler mirror not found.
[error] scala.reflect.internal.MissingRequirementError$.notFound(MissingRequirementError.scala:24)
[error] scala.reflect.internal.Mirrors$RootsBase.$anonfun$getModuleOrClass$6(Mirrors.scala:66)
[error] scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:66)
[error] scala.reflect.internal.Mirrors$RootsBase.getRequiredClass(Mirrors.scala:56)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$JUnitAnnots$.<init>(ScalaNativeJUnitPlugin.scala:48)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$.JUnitAnnots$lzycompute$1(ScalaNativeJUnitPlugin.scala:47)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$.scala$scalanative$junit$plugin$ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$$JUnitAnnots(ScalaNativeJUnitPlugin.scala:47)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer.$anonfun$transform$1(ScalaNativeJUnitPlugin.scala:97)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer.$anonfun$transform$1$adapted(ScalaNativeJUnitPlugin.scala:96)
[error] scala.collection.IterableOnceOps.exists(IterableOnce.scala:591)
[error] scala.collection.IterableOnceOps.exists$(IterableOnce.scala:588)
[error] scala.collection.AbstractIterable.exists(Iterable.scala:926)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer.hasTests$1(ScalaNativeJUnitPlugin.scala:96)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer.scala$scalanative$junit$plugin$ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer$$isTest$1(ScalaNativeJUnitPlugin.scala:107)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer$$anonfun$1.applyOrElse(ScalaNativeJUnitPlugin.scala:111)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer$$anonfun$1.applyOrElse(ScalaNativeJUnitPlugin.scala:110)
[error] scala.collection.immutable.List.collect(List.scala:267)
[error] scala.scalanative.junit.plugin.ScalaNativeJUnitPlugin$ScalaNativeJUnitPluginComponent$ScalaNativeJUnitPluginTransformer.transform(ScalaNativeJUnitPlugin.scala:110)
[error] scala.tools.nsc.ast.Trees$Transformer.transformUnit(Trees.scala:183)
[error] scala.tools.nsc.transform.Transform$Phase.apply(Transform.scala:32)
[error] scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:459)
[error] scala.tools.nsc.Global$GlobalPhase.run(Global.scala:406)
[error] scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1524)
[error] scala.tools.nsc.Global$Run.compileUnits(Global.scala:1508)
[error] scala.tools.nsc.Global$Run.compileSources(Global.scala:1500)
[error] scala.tools.nsc.Global$Run.compileFiles(Global.scala:1613)
[error] xsbt.CachedCompiler0.run(CompilerBridge.scala:163)
```

Comment (xuwei-k) [reply build.sbt:125]: and there is no easy way `add Scala compiler plugin only for test sources`

https://stackoverflow.com/questions/62890985

File: build.sbt [hunk 3]
```diff
@@ -121,6 +121,21 @@ def configureJUnit(crossProject: CrossProject) = {
...
+    addCompilerPlugin("org.scala-native" % "junit-plugin" % nativeVersion cross CrossVersion.full),
+    pomPostProcess := { node =>
+      import scala.xml._
+      import scala.xml.transform._
+      new RuleTransformer(new RewriteRule{
+        override def transform(n: Node) =
+          if (n.label == "dependency" && (n \ "artifactId").text.startsWith("junit-runtime_native0.4"))
```
Comment (joroKr21) [inline build.sbt:132]: Let's not bake in the version, WDYT?
```suggestion
          if (n.label == "dependency" && (n \ "artifactId").text.startsWith("junit-runtime_native"))
```

File: build.sbt (hunk 2 above)
Comment (joroKr21) [reply build.sbt:125]: Hmm I noticed that it works in shapeless-3 - how is that possible? 🤔

**Pull request merged**